src/kademlia/get_item.cpp

```diff
@@ -78,7 +78,7 @@ void get_item::got_data(bdecode_node const& v,
 	// data can reach here, which means pk and sig must be valid.
 	if (!pk || !sig) return;
 
-	std::pair<char const*, int> salt(m_salt.c_str(), int(m_salt.size()));
```

Comment:
I'd like to see a copy of the salt string made here. As is, the salt string gets assigned to itself down in `m_data.assign`. While this probably wouldn't cause a problem in practice, I think it technically invokes Undefined Behavior and the aliasing here makes me nervous.